# U.S. Medical Insurance Costs

### Project questions:

* What is the insurance charge average by gender?
* Is there a difference between women with children vs man with children?
* What characteristics affects the cost for medical insurance?
* What make the insurance cost higher?
* What region have the highest insurance charge? And why?
* Which age group buy more insurance?

In [1]:
import csv


In [2]:
class Insured:
    def __init__(self, age, gender, bmi, children, smoker, region, charges):
        self.age = int(age)
        self.gender = gender
        self.bmi = round(float(bmi), 1)
        self.children = int(children)
        self.smoker = True if smoker == "yes" else False
        self.region = region
        self.charges = float(charges)

    def __repr__(self):
        return f"""
        Age: {self.age}
        Gender: {self.gender}
        BMI: {self.bmi}
        Number of children: {self.children}
        Smoker status: {self.smoker}
        Region: {self.region}
        Insurance charges: ${round(self.charges, 2)}
        """


In [3]:
class Insights:
    def __init__(self, insured_list):
        self.insured_list = insured_list
        self.build_dict_by_gender()
        self.build_dict_by_smoker()
        self.build_dict_by_region()
        self.build_dict_by_age()
        self.build_dict_by_bmi()

    def build_dict_by_gender(self):
        self.insured_by_gender = {}
        for insured in self.insured_list:
            if insured.gender not in self.insured_by_gender:
                self.insured_by_gender[insured.gender] = [insured]
            else:
                self.insured_by_gender[insured.gender].append(insured)

    def build_dict_by_smoker(self):
        self.insured_by_smoker = {}
        for insured in self.insured_list:
            if insured.smoker not in self.insured_by_smoker:
                self.insured_by_smoker[insured.smoker] = [insured]
            else:
                self.insured_by_smoker[insured.smoker].append(insured)

    def build_dict_by_region(self):
        self.insured_by_region = {}
        for insured in self.insured_list:
            if insured.region not in self.insured_by_region:
                self.insured_by_region[insured.region] = [insured]
            else:
                self.insured_by_region[insured.region].append(insured)

    def build_dict_by_age(self):
        self.insured_by_age = {}
        for insured in self.insured_list:
            key = "Under 18"
            if insured.age >= 18 and insured.age < 25:
                key = "18-24"
            elif insured.age >= 25 and insured.age < 35:
                key = "25-34"
            elif insured.age >= 35 and insured.age < 45:
                key = "35-44"
            elif insured.age >= 45 and insured.age < 55:
                key = "45-54"
            elif insured.age >= 55 and insured.age < 65:
                key = "55-64"
            elif insured.age >= 65:
                key = "65 or older"
            if key not in self.insured_by_age:
                self.insured_by_age[key] = [insured]
            else:
                self.insured_by_age[key].append(insured)

    def build_dict_by_bmi(self):
        self.insured_by_bmi = {}
        for insured in self.insured_list:
            key = "Underweight"
            if insured.bmi >= 18.5 and insured.bmi < 25.0:
                key = "Normal weight"
            elif insured.bmi >= 25.0 and insured.bmi < 30.0:
                key = "Overweight"
            elif insured.bmi >= 30.0 and insured.bmi < 35.0:
                key = "Obesity first degree"
            elif insured.bmi >= 35.0 and insured.bmi < 40.0:
                key = "Obesity second degree"
            elif insured.bmi >= 40.0:
                key = "Obesity third degree"
            if key not in self.insured_by_bmi:
                self.insured_by_bmi[key] = [insured]
            else:
                self.insured_by_bmi[key].append(insured)

    def filter_by_children(self, list):
        insured_by_children = {}
        for insured in list:
            key = "With children"
            if insured.children == 0:
                key = "Without children"
            if key not in insured_by_children:
                insured_by_children[key] = [insured]
            else:
                insured_by_children[key].append(insured)
        return insured_by_children

    def calculate_bmi_average(self, list):
        total_of_bmi = 0
        for insured in list:
            total_of_bmi += insured.bmi
        return round(total_of_bmi / len(list), 2)

    def calculate_average_children(self, list):
        total_of_children = 0
        for insured in list:
            total_of_children += insured.children
        return round(total_of_children / len(list), 0)

    def calculate_average_cost(self, list):
        total_cost = 0
        for insured in list:
            total_cost += insured.charges
        return round(total_cost / len(list), 2)

    def calculate_average_age(self, list):
        total_age = 0
        for insured in list:
            total_age += insured.age
        return round(total_age / len(list), 2)

    def calculate_difference(self, value1, value2):
        return abs(round(((value1 / value2) * 100) - 100, 0))

    def calculate_percentage(self, value1, value2):
        return abs(round((value1 / value2) * 100, 0))


In [4]:
with open("insurance.csv") as csv_file:
    insurance_file = csv.DictReader(csv_file)

    insured_list = []
    for row in insurance_file:
        insured_list.append(
            Insured(
                row["age"],
                row["sex"],
                row["bmi"],
                row["children"],
                row["smoker"],
                row["region"],
                row["charges"],
            )
        )


In [5]:
insight = Insights(insured_list)


### Gender metrics

In [6]:
total_of_women = len(insight.insured_by_gender["female"])
total_of_men = len(insight.insured_by_gender["male"])
total_of_insured = len(insight.insured_list)

percentage_men = insight.calculate_percentage(total_of_men, total_of_insured)

print(
    f"There are {total_of_women} women and {total_of_men} men insured. Men represent {percentage_men}% of all insured."
)


There are 662 women and 676 men insured. Men represent 51.0% of all insured.


### Average of insurance charge by gender

In [7]:
insurance_average = insight.calculate_average_cost(insight.insured_list)
insurance_average_women = insight.calculate_average_cost(
    insight.insured_by_gender["female"]
)
insurance_average_men = insight.calculate_average_cost(
    insight.insured_by_gender["male"]
)
percentage = insight.calculate_difference(
    insurance_average_men, insurance_average_women
)

print(f"The overall average of insurance charge is: $ {insurance_average} dollars.")
print(
    f"The average of insurance charge for women is: ${insurance_average_women} dollars."
)
print(f"The average of insurance charge for men is: ${insurance_average_men} dollars.")
print(
    f"If we analyze only gender, men pay about {percentage}% more than women for medical insurance."
)


The overall average of insurance charge is: $ 13270.42 dollars.
The average of insurance charge for women is: $12569.58 dollars.
The average of insurance charge for men is: $13956.75 dollars.
If we analyze only gender, men pay about 11.0% more than women for medical insurance.


### Parents metrics

In [8]:
people_with_children = len(
    insight.filter_by_children(insight.insured_list)["With children"]
)
women_without_children = len(
    insight.filter_by_children(insight.insured_by_gender["female"])["Without children"]
)
men_without_children = len(
    insight.filter_by_children(insight.insured_by_gender["male"])["Without children"]
)
parent_percentage = insight.calculate_percentage(women_without_children, total_of_women)
women_percentage = insight.calculate_percentage(people_with_children, total_of_insured)
men_percentage = insight.calculate_percentage(men_without_children, total_of_men)

print(
    f"There are {people_with_children} parents and this represents {parent_percentage}% of all insured."
)
print(
    f"There are {women_without_children} women without children and this represents {women_percentage}% of all women."
)
print(
    f"There are {men_without_children} men without children and this represents {men_percentage}% of all men."
)


There are 764 parents and this represents 44.0% of all insured.
There are 289 women without children and this represents 57.0% of all women.
There are 285 men without children and this represents 42.0% of all men.


### Average of children by gender

In [9]:
average_children_for_women = insight.calculate_average_children(
    insight.insured_by_gender["female"]
)
average_children_for_men = insight.calculate_average_children(
    insight.insured_by_gender["male"]
)

print(f"The average number of children for women are: {average_children_for_women}.")
print(f"The average number of children for men are: {average_children_for_men}.")


The average number of children for women are: 1.0.
The average number of children for men are: 1.0.


### Average charge for parents

In [10]:
average_cost_parents = insight.calculate_average_cost(
    insight.filter_by_children(insight.insured_list)["With children"]
)
average_cost_without_children = insight.calculate_average_cost(
    insight.filter_by_children(insight.insured_list)["Without children"]
)
percentage_parents = insight.calculate_difference(
    average_cost_parents, average_cost_without_children
)

print(
    f"The average insurance charge for people with children is ${average_cost_parents} dollars, that is {percentage_parents}% more than people without children."
)
print(
    f"The average for people without children is: ${average_cost_without_children} dollars."
)


The average insurance charge for people with children is $13949.94 dollars, that is 13.0% more than people without children.
The average for people without children is: $12365.98 dollars.


### Smoker status metrics

In [11]:
smokers = list(filter(lambda insured: insured.smoker, insight.insured_list))
percentage_smokers = insight.calculate_percentage(len(smokers), len(insured_list))
print(f" Smokers represent {percentage_smokers}% of all insured.")


 Smokers represent 20.0% of all insured.


### Average of insurance charge by smoker status

In [12]:
insurance_average_smoker = insight.calculate_average_cost(
    insight.insured_by_smoker[True]
)
insurance_average_non_smoker = insight.calculate_average_cost(
    insight.insured_by_smoker[False]
)
percentage = insight.calculate_difference(
    insurance_average_smoker, insurance_average_non_smoker
)

print(
    f"The average of insurance charge for a person who smokes is: ${insurance_average_smoker} dollars."
)
print(
    f"The average of insurance charge for someone who does not smoke is: ${insurance_average_non_smoker} dollars."
)
print(
    f"If we analyze only the smoker status, a smoker pays {percentage}% more than a non smoker."
)


The average of insurance charge for a person who smokes is: $32050.23 dollars.
The average of insurance charge for someone who does not smoke is: $8434.27 dollars.
If we analyze only the smoker status, a smoker pays 280.0% more than a non smoker.


### Percentage of smoker by gender

In [13]:
smoker_women = list(
    filter(lambda insured: insured.smoker, insight.insured_by_gender["female"])
)
smoker_men = list(
    filter(lambda insured: insured.smoker, insight.insured_by_gender["male"])
)
percentage_smoker_women = insight.calculate_percentage(
    len(smoker_women), total_of_women
)
percentage_smoker_men = insight.calculate_percentage(len(smoker_men), total_of_men)
average_cost_smoker_women = insight.calculate_average_cost(smoker_women)
average_cost_smoker_men = insight.calculate_average_cost(smoker_men)

print(
    f"Smoker women represent {percentage_smoker_women}% of all women. Their average insurance charge is ${average_cost_smoker_women} dollars."
)
print(
    f"Smoker men represent {percentage_smoker_men}% of all men. Their average insurance charge is ${average_cost_smoker_men} dollars."
)


Smoker women represent 17.0% of all women. Their average insurance charge is $30679.0 dollars.
Smoker men represent 24.0% of all men. Their average insurance charge is $33042.01 dollars.


### Percentage of non smoker by gender

In [14]:
non_smoker_women = list(
    filter(lambda insured: not insured.smoker, insight.insured_by_gender["female"])
)
non_smoker_men = list(
    filter(lambda insured: not insured.smoker, insight.insured_by_gender["male"])
)
percentage_nonsmoker_women = insight.calculate_percentage(
    len(non_smoker_women), total_of_women
)
percentage_nonsmoker_men = insight.calculate_percentage(
    len(non_smoker_men), total_of_men
)
average_cost_nonsmoker_women = insight.calculate_average_cost(non_smoker_women)
average_cost_nonsmoker_men = insight.calculate_average_cost(non_smoker_men)
difference_cost = insight.calculate_difference(
    average_cost_nonsmoker_women, average_cost_nonsmoker_men
)

print(
    f"Non smoker women represent {percentage_nonsmoker_women}% of all women. Their average insurance charge is ${average_cost_nonsmoker_women} dollars."
)
print(
    f"Non smoker men represent {percentage_nonsmoker_men}% of all men. Their average insurance charge is ${average_cost_nonsmoker_men} dollars."
)
print(f"Women pay {difference_cost}% more than men, between non smokers.")


Non smoker women represent 83.0% of all women. Their average insurance charge is $8762.3 dollars.
Non smoker men represent 76.0% of all men. Their average insurance charge is $8087.2 dollars.
Women pay 8.0% more than men, between non smokers.


### Region metrics

In [15]:
insight.insured_by_region.keys()


dict_keys(['southwest', 'southeast', 'northwest', 'northeast'])

In [16]:
insured_from_southeast = insight.calculate_percentage(
    len(insight.insured_by_region["southeast"]), len(insight.insured_list)
)
insured_from_southwest = insight.calculate_percentage(
    len(insight.insured_by_region["southwest"]), len(insight.insured_list)
)
insured_from_northeast = insight.calculate_percentage(
    len(insight.insured_by_region["northeast"]), len(insight.insured_list)
)
insured_from_northwest = insight.calculate_percentage(
    len(insight.insured_by_region["northwest"]), len(insight.insured_list)
)

print(
    f"The residents of Southeast region represent {insured_from_southeast}% of the insured."
)
print(
    f"The residents of Southwest region represent {insured_from_southwest}% of the insured."
)
print(
    f"The residents of Northeast region represent {insured_from_northeast}% of the insured."
)
print(
    f"The residents of Northwest region represent {insured_from_northwest}% of the insured."
)


The residents of Southeast region represent 27.0% of the insured.
The residents of Southwest region represent 24.0% of the insured.
The residents of Northeast region represent 24.0% of the insured.
The residents of Northwest region represent 24.0% of the insured.


### Average by region

In [17]:
northwest_average = insight.calculate_average_cost(
    insight.insured_by_region["northwest"]
)
northeast_average = insight.calculate_average_cost(
    insight.insured_by_region["northeast"]
)
southwest_average = insight.calculate_average_cost(
    insight.insured_by_region["southwest"]
)
southeast_average = insight.calculate_average_cost(
    insight.insured_by_region["southeast"]
)
print(f"The average cost for the Northwest region is ${northwest_average}.")
print(f"The average cost for the Northeast region is ${northeast_average}.")
print(f"The average cost for the Southwest region is ${southwest_average}.")
print(f"The average cost for the Southeast region is ${southeast_average}.")


The average cost for the Northwest region is $12417.58.
The average cost for the Northeast region is $13406.38.
The average cost for the Southwest region is $12346.94.
The average cost for the Southeast region is $14735.41.


### Smokers presence per region

In [18]:
smokers_southeast = list(
    filter(lambda insured: insured.smoker, insight.insured_by_region["southeast"])
)
smokers_northeast = list(
    filter(lambda insured: insured.smoker, insight.insured_by_region["northeast"])
)
smokers_southwest = list(
    filter(lambda insured: insured.smoker, insight.insured_by_region["southwest"])
)
smokers_northwest = list(
    filter(lambda insured: insured.smoker, insight.insured_by_region["northwest"])
)
percentage_smokers_southeast = insight.calculate_percentage(
    len(smokers_southeast), len(insight.insured_by_region["southeast"])
)
percentage_smokers_northeast = insight.calculate_percentage(
    len(smokers_northeast), len(insight.insured_by_region["northeast"])
)
percentage_smokers_southwest = insight.calculate_percentage(
    len(smokers_southwest), len(insight.insured_by_region["southwest"])
)
percentage_smokers_northwest = insight.calculate_percentage(
    len(smokers_northwest), len(insight.insured_by_region["northwest"])
)
print(f"Smokers represent {percentage_smokers_southeast}% of the Southeast region.")
print(f"Smokers represent {percentage_smokers_northeast}% of the Northeast region.")
print(f"Smokers represent {percentage_smokers_southwest}% of the Southwest region.")
print(f"Smokers represent {percentage_smokers_northwest}% of the Northwest region.")


Smokers represent 25.0% of the Southeast region.
Smokers represent 21.0% of the Northeast region.
Smokers represent 18.0% of the Southwest region.
Smokers represent 18.0% of the Northwest region.


### Age group metrics

Distributed in:

* Under 18 (empty/no key was made)
* 18 - 24
* 25 - 34
* 35 - 44
* 45 - 54
* 55 - 64
* 65 and older (empty/no key was made)

In [19]:
insight.insured_by_age.keys()


dict_keys(['18-24', '25-34', '45-54', '35-44', '55-64'])

In [20]:
age_average = insight.calculate_average_age(insight.insured_list)
percentage_for_18_24 = insight.calculate_percentage(
    len(insight.insured_by_age["18-24"]), total_of_insured
)
percentage_for_25_34 = insight.calculate_percentage(
    len(insight.insured_by_age["25-34"]), total_of_insured
)
percentage_for_35_44 = insight.calculate_percentage(
    len(insight.insured_by_age["35-44"]), total_of_insured
)
percentage_for_45_54 = insight.calculate_percentage(
    len(insight.insured_by_age["45-54"]), total_of_insured
)
percentage_for_55_64 = insight.calculate_percentage(
    len(insight.insured_by_age["55-64"]), total_of_insured
)

print(f"The age average of all the insured is {age_average} years old.")
print(f"The 18-24 age group represent {percentage_for_18_24}% of all insured.")
print(f"The 25-34 age group represent {percentage_for_25_34}% of all insured.")
print(f"The 35-44 age group represent {percentage_for_35_44}% of all insured.")
print(f"The 45-54 age group represent {percentage_for_45_54}% of all insured.")
print(f"The 55-64 age group represent {percentage_for_55_64}% of all insured.")


The age average of all the insured is 39.21 years old.
The 18-24 age group represent 21.0% of all insured.
The 25-34 age group represent 20.0% of all insured.
The 35-44 age group represent 19.0% of all insured.
The 45-54 age group represent 21.0% of all insured.
The 55-64 age group represent 18.0% of all insured.


### Average charge per age group

In [21]:
average_cost_18_24 = insight.calculate_average_cost(insight.insured_by_age["18-24"])
average_cost_25_34 = insight.calculate_average_cost(insight.insured_by_age["25-34"])
average_cost_35_44 = insight.calculate_average_cost(insight.insured_by_age["35-44"])
average_cost_45_54 = insight.calculate_average_cost(insight.insured_by_age["45-54"])
average_cost_55_64 = insight.calculate_average_cost(insight.insured_by_age["55-64"])

print(f"The average charge for group age of 18-24 is: ${average_cost_18_24} dollars.")
print(f"The average charge for group age of 25-34 is: ${average_cost_25_34} dollars.")
print(f"The average charge for group age of 35-44 is: ${average_cost_35_44} dollars.")
print(f"The average charge for group age of 45-54 is: ${average_cost_45_54} dollars.")
print(f"The average charge for group age of 55-64 is: ${average_cost_55_64} dollars.")


The average charge for group age of 18-24 is: $9011.34 dollars.
The average charge for group age of 25-34 is: $10352.39 dollars.
The average charge for group age of 35-44 is: $13134.17 dollars.
The average charge for group age of 45-54 is: $15853.93 dollars.
The average charge for group age of 55-64 is: $18513.28 dollars.


### Age average by region

In [22]:
southeast_age_average = insight.calculate_average_age(
    insight.insured_by_region["southeast"]
)
northeast_age_average = insight.calculate_average_age(
    insight.insured_by_region["northeast"]
)
southwest_age_average = insight.calculate_average_age(
    insight.insured_by_region["southwest"]
)
northwest_age_average = insight.calculate_average_age(
    insight.insured_by_region["northwest"]
)

print(f"The average age for Southeast region is: {southeast_age_average}.")
print(f"The average age for Northeast region is: {northeast_age_average}.")
print(f"The average age for Southwest region is: {southwest_age_average}.")
print(f"The average age for Northwest region is: {northwest_age_average}.")


The average age for Southeast region is: 38.94.
The average age for Northeast region is: 39.27.
The average age for Southwest region is: 39.46.
The average age for Northwest region is: 39.2.


### BMI metrics

Distributed in:

* Underweight: less than 18.5
* Normal weight: 18.5 - 24,9
* Overweight: 25.0 - 29,9
* Obesity first degree: 30.0 - 34,9
* Obesity second degree: 35.0 - 39.9
* Obesity third degree: more than 40.0

In [23]:
insight.insured_by_bmi.keys()


dict_keys(['Overweight', 'Obesity first degree', 'Normal weight', 'Obesity second degree', 'Obesity third degree', 'Underweight'])

In [24]:
percentage_for_underweight = insight.calculate_percentage(
    len(insight.insured_by_bmi["Underweight"]), total_of_insured
)
percentage_for_normalweight = insight.calculate_percentage(
    len(insight.insured_by_bmi["Normal weight"]), total_of_insured
)
percentage_for_overweight = insight.calculate_percentage(
    len(insight.insured_by_bmi["Overweight"]), total_of_insured
)
percentage_for_obesity_first_degree = insight.calculate_percentage(
    len(insight.insured_by_bmi["Obesity first degree"]), total_of_insured
)
percentage_for_obesity_second_degree = insight.calculate_percentage(
    len(insight.insured_by_bmi["Obesity second degree"]), total_of_insured
)
percentage_for_obesity_third_degree = insight.calculate_percentage(
    len(insight.insured_by_bmi["Obesity third degree"]), total_of_insured
)

print(
    f"People who is in the underweight range represents {percentage_for_underweight}% of all insured."
)
print(
    f"People who is in the normal weight range represents {percentage_for_normalweight}% of all insured."
)
print(
    f"People who is in the overweight range represents {percentage_for_overweight}% of all insured."
)
print(
    f"People who is in the first degree obesity range represents {percentage_for_obesity_first_degree}% of all insured."
)
print(
    f"People who is in the second degree obesity range represents {percentage_for_obesity_second_degree}% of all insured."
)
print(
    f"People who is in the third degree obesity range represents {percentage_for_obesity_third_degree}% of all insured."
)


People who is in the underweight range represents 1.0% of all insured.
People who is in the normal weight range represents 17.0% of all insured.
People who is in the overweight range represents 29.0% of all insured.
People who is in the first degree obesity range represents 29.0% of all insured.
People who is in the second degree obesity range represents 17.0% of all insured.
People who is in the third degree obesity range represents 7.0% of all insured.


### Average bmi per all, gender, region

In [25]:
bmi_average = insight.calculate_bmi_average(insight.insured_list)
bmi_women = insight.calculate_bmi_average(insight.insured_by_gender["female"])
bmi_men = insight.calculate_bmi_average(insight.insured_by_gender["male"])
bmi_southeast = insight.calculate_bmi_average(insight.insured_by_region["southeast"])
bmi_northeast = insight.calculate_bmi_average(insight.insured_by_region["northeast"])
bmi_southwest = insight.calculate_bmi_average(insight.insured_by_region["southwest"])
bmi_northwest = insight.calculate_bmi_average(insight.insured_by_region["northwest"])

print(f"The average bmi for all insured is: {bmi_average}.")
print(f"The average bmi for all insured women is: {bmi_women}.")
print(f"The average bmi for all insured men is: {bmi_men}.")
print(f"The average bmi for Southeast region is: {bmi_southeast}.")
print(f"The average bmi for Northeast region is: {bmi_northeast}.")
print(f"The average bmi for Southwest region is: {bmi_southwest}.")
print(f"The average bmi for Northwest region is: {bmi_northwest}.")


The average bmi for all insured is: 30.66.
The average bmi for all insured women is: 30.38.
The average bmi for all insured men is: 30.94.
The average bmi for Southeast region is: 33.36.
The average bmi for Northeast region is: 29.17.
The average bmi for Southwest region is: 30.6.
The average bmi for Northwest region is: 29.2.


### Average bmi per age group

In [26]:
average_bmi_18_24 = insight.calculate_bmi_average(insight.insured_by_age["18-24"])
average_bmi_25_34 = insight.calculate_bmi_average(insight.insured_by_age["25-34"])
average_bmi_35_44 = insight.calculate_bmi_average(insight.insured_by_age["35-44"])
average_bmi_45_54 = insight.calculate_bmi_average(insight.insured_by_age["45-54"])
average_bmi_55_64 = insight.calculate_bmi_average(insight.insured_by_age["55-64"])

print(
    f"The average bmi for people between 18 and 24 years old is: {average_bmi_18_24}."
)
print(
    f"The average bmi for people between 25 and 34 years old is: {average_bmi_25_34}."
)
print(
    f"The average bmi for people between 35 and 44 years old is: {average_bmi_35_44}."
)
print(
    f"The average bmi for people between 45 and 54 years old is: {average_bmi_45_54}."
)
print(
    f"The average bmi for people between 55 and 64 years old is: {average_bmi_55_64}."
)


The average bmi for people between 18 and 24 years old is: 30.04.
The average bmi for people between 25 and 34 years old is: 30.06.
The average bmi for people between 35 and 44 years old is: 30.4.
The average bmi for people between 45 and 54 years old is: 31.15.
The average bmi for people between 55 and 64 years old is: 31.76.


### Average charge by bmi group

In [27]:
average_cost_underweight = insight.calculate_average_cost(
    insight.insured_by_bmi["Underweight"]
)
average_cost_normalweight = insight.calculate_average_cost(
    insight.insured_by_bmi["Normal weight"]
)
average_cost_overweight = insight.calculate_average_cost(
    insight.insured_by_bmi["Overweight"]
)
average_cost_obesity_first_degree = insight.calculate_average_cost(
    insight.insured_by_bmi["Obesity first degree"]
)
average_cost_obesity_second_degree = insight.calculate_average_cost(
    insight.insured_by_bmi["Obesity second degree"]
)
average_cost_obesity_third_degree = insight.calculate_average_cost(
    insight.insured_by_bmi["Obesity third degree"]
)

print(
    f"The average insurance charge for a underweight person is ${average_cost_underweight} dollars."
)
print(
    f"The average insurance charge for a normal weight person is ${average_cost_normalweight} dollars."
)
print(
    f"The average insurance charge for a overweight person is ${average_cost_overweight} dollars."
)
print(
    f"The average insurance charge for a person with first degree obesity is ${average_cost_obesity_first_degree} dollars."
)
print(
    f"The average insurance charge for a person with second degree obesity is ${average_cost_obesity_second_degree} dollars."
)
print(
    f"The average insurance charge for a person with third degree obesity is ${average_cost_obesity_third_degree} dollars."
)


The average insurance charge for a underweight person is $8852.2 dollars.
The average insurance charge for a normal weight person is $10379.5 dollars.
The average insurance charge for a overweight person is $11000.08 dollars.
The average insurance charge for a person with first degree obesity is $14298.84 dollars.
The average insurance charge for a person with second degree obesity is $17226.63 dollars.
The average insurance charge for a person with third degree obesity is $16707.66 dollars.


## Insights



* It has almost the same number of male and female insured, about 1% more of men.
* The age group is well distributed. Each group age represent 18% to 20% of the insured.
* People who have at least 1 child represent about 44% of the insured.
* There are more male parent than women: 57% of women and 42% of men do not have children.
* The average bmi is about 31.0 and more than 80% of all insured are overweighted.
* Smokers represent 20% of the insured. 
* There are more resident of Southeast among the insured. They correspond to 27% of the insured.

* If we analyze the average cost, only considering gender, men would pay 11% more of insurance charge.
* Number of children does not seem to be the cause for the difference of insurance charge between men and women.
* It is possible to observe that smoker status influence more than anything the cost of insurance. Smoker tends to pay 280% more than non smokers.
* Age, gender, bmi, number of children contribute for increasing the insurance cost.
* Parents pay 13% more for insurance than people without children.
* If we analyze only non smokers, women pay 8% more for insurance than men. 
* The residents of Southeast region pay more for insurance. Although the region per se does not affect the cost, the insured in this region are heavier and smoke more.
* It was not possible to notice if a specific group age is more preoccupied of being insured than other groups. 